test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -229,10 +229,15 @@ bool UcxContext::listen(const struct sockaddr* saddr, size_t addrlen)
     return true;
 }
 
-void UcxContext::progress()
+void UcxContext::progress(unsigned count)
 {
-    progress_worker_event();
-    progress_io_message();
+    unsigned i = 1;
+    bool ret;
+
+    do {
```

Comment:
```
for (i = 0; i < count; ++i) {
  if (!progress_worker_event()) {
    break;
  }

  progress_io_message();
}
```

Reply:
in epoll case, progress_worker_event() returning false could mean having made progress via epoll.

Reply:
shouldn't we call progress_io_message() anyway regardless what progress_worker_event() returns?

Reply:
if progress_worker_event returned false, it means ucp_worker_progress in it returned false, so nothing was received

Reply:
that's right when epoll is disabled. when epoll is enabled, returning false would mean having made progress via epoll.

Reply:
epoll_wait does not really progress. it just yields the CPU until some UCX event may have happened.
i don't see that ucp_worker_progress() is called again after epoll

Reply:
correct. i thought it in a wrong way. will commit a new patch after back to work. on a vacation now.